### PR TITLE
resolves #18, mitigate variable referenced before assignment error if…

### LINKED ISSUE #18: "UnboundLocalError: local variable 'response' referenced before assignment" on Timeout exception
The code below will fail for "UnboundLocalError: local variable 'response' referenced before assignment" at line "if response.ok:" if Timeout exception occurs since response variable is not initialized.

try:
    response = requests.post(self._url, headers=headers, data=body,
                             timeout=(KUMO_CONNECT_TIMEOUT_SECONDS,
                                      KUMO_RESPONSE_TIMEOUT_SECONDS))
except Timeout as ex:
    _LOGGER.warning("Timeout querying KumoCloud: %s", str(ex))
if response.ok:
    self._kumo_dict = response.json()

### PR DESCRIPTION
**Pull request by cjkrolak**
… Timeout exception causes response to be undefined

### CODE REVIEW

**Comment by cjkrolak** [comment]
before fix:
`Timeout querying KumoCloud: HTTPSConnectionPool(host='geo-c.kumocloud.com', port=443): Read timed out. (read timeout=0.1)
Traceback (most recent call last):
  File "C:\Users\cjkro\onedrive\documents\github\ThermostatSupervisor\kumolocal.py", line 571, in <module>
    Thermostat, Zone = tc.thermostat_basic_checkout(
  File "C:\Users\cjkro\onedrive\documents\github\ThermostatSupervisor\thermostat_common.py", line 953, in thermostat_basic_checkout
    Thermostat, Zone = create_thermostat_instance(api, thermostat_type, zone,
  File "C:\Users\cjkro\onedrive\documents\github\ThermostatSupervisor\thermostat_common.py", line 921, in create_thermostat_instance
    Thermostat = ThermostatClass(zone)
  File "C:\Users\cjkro\onedrive\documents\github\ThermostatSupervisor\kumolocal.py", line 53, in __init__
    self.device_id = self.get_target_zone_id(self.zone_number)
  File "C:\Users\cjkro\onedrive\documents\github\ThermostatSupervisor\kumolocal.py", line 68, in get_target_zone_id
    kumos = self.make_pykumos()
  File "..\pykumo\pykumo\pykumo.py", line 625, in make_pykumos
    for iu in list(self.get_indoor_units()):
  File "..\pykumo\pykumo\pykumo.py", line 562, in get_indoor_units
    self._fetch_if_needed()
  File "..\pykumo\pykumo\pykumo.py", line 522, in _fetch_if_needed
    if response.ok:
UnboundLocalError: local variable 'response' referenced before assignment`

after fix:
`Timeout querying KumoCloud: HTTPSConnectionPool(host='geo-c.kumocloud.com', port=443): Read timed out. (read timeout=0.1)
No JSON returned from KumoCloud; check credentials?`

File: pykumo/pykumo.py [hunk 1]
```diff
@@ -516,11 +516,12 @@ def _fetch_if_needed(self):
                                                   KUMO_RESPONSE_TIMEOUT_SECONDS))
             except Timeout as ex:
                 _LOGGER.warning("Timeout querying KumoCloud: %s", str(ex))
-            if response.ok:
-                self._kumo_dict = response.json()
-            else:
-                _LOGGER.warning("Error response from KumoCloud: %s %s}",
-                                str(response.status_code), response.text)
+            if 'response' in locals():
```
Comment (dlarrick) [inline pykumo/pykumo.py:519]: Thanks... stylistically I'd prefer assigning `response=None` in the `except` block and testing `if response`.

**Comment by dlarrick** [comment]
I've included the equivalent of this fix in https://github.com/dlarrick/pykumo/pull/25 and will close this PR and the issue when that's merged.

**Comment by dlarrick** [comment]
Thanks, I incorporated this change into 0.2.0

**Pull request closed without merging**